core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -277,27 +277,62 @@ func (r *EvmRegistry) refreshActiveUpkeeps() error {
 //
 // TODO: check for updated config for log trigger upkeeps and update it, currently we ignore them.
 func (r *EvmRegistry) refreshLogTriggerUpkeeps(ids []*big.Int) error {
```

Comment:
We are doing some serious work here on the entire set of log upkeeps that we have in the system, which is not a scalable solution. You can break it into multiple batches (batch size can be 32)

Reply:
So, use a worker pool to process these upkeep IDs in batches of 32 at a time?

Reply:
We shouldn't use errgroup in this case, see the docs [docs](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go):
`The first call to return a non-nil error cancels the group's context`

In general, fanning out won't balance the workload on the other components.
One simple option is to run on the same goroutine and distribute the work over time, 
i.e. call `r.refreshLogTriggerUpkeepsBatch(batch)` in the same goroutine and wait after each batch

Reply:
Oh so rather than trying to parallelize it, we should be staggering the operation into smaller batches instead?